pkg/operator/vspherecontroller/vsphere_environment_checker.go

```diff
@@ -37,15 +38,17 @@ type vSphereEnvironmentCheckerComposite struct {
 // make sure that vSphereEnvironmentCheckerComposite implements the vSphereEnvironmentCheckInterface
 var _ vSphereEnvironmentCheckInterface = &vSphereEnvironmentCheckerComposite{}
 
-func newVSphereEnvironmentChecker() *vSphereEnvironmentCheckerComposite {
+func newVSphereEnvironmentChecker(clusterCSIDriverLister opv1.ClusterCSIDriverLister) *vSphereEnvironmentCheckerComposite {
 	checker := &vSphereEnvironmentCheckerComposite{
 		backoff:   defaultBackoff,
 		nextCheck: time.Now(),
 	}
 	checker.checkers = []checks.CheckInterface{
 		&checks.CheckExistingDriver{},
 		&checks.VCenterChecker{},
-		&checks.NodeChecker{},
+		&checks.NodeChecker{
+			ClusterCSIDriverLister: clusterCSIDriverLister,
+		},
```

Comment:
So, rather than passing this as one-off, shouldn't this be part of - https://github.com/openshift/vmware-vsphere-csi-driver-operator/blob/main/pkg/operator/vspherecontroller/checks/api_interface.go#L27 ? That will keep everything uniform between checkers.

Reply:
Right, it should - changed.